code/modules/mob/living/status_procs.dm

```diff
@@ -616,6 +647,45 @@
 			P = apply_status_effect(STATUS_EFFECT_WEAKENED, amount)
 	return P
 
+/mob/living/proc/IsKnockedDown()
+	return has_status_effect(STATUS_EFFECT_FLOORED)
+
+/mob/living/proc/AmountKnockDown()
+	var/datum/status_effect/incapacitating/floored/F = IsKnockedDown()
+	if(F)
+		return F.duration - world.time
```

Comment:
just a minor concern here, this will perform the math operation before returning `F.duration - world.time` correct? I remember with defines having to surround operations with parentheses for that reason but I don't know if that logic would apply here or not.

Reply:
the same operation happens in other places in this file. this is valid syntax